lib/kubereq.ex

```diff
@@ -97,13 +111,60 @@ defmodule Kubereq do
   @spec new(kubeconfig :: Kubereq.Kubeconfig.t(), resource_path :: binary()) ::
           Req.Request.t()
   def new(kubeconfig, resource_path) do
-    new(kubeconfig)
-    |> Req.merge(
+    Req.new()
+    |> attach(kubeconfig: kubeconfig)
+    |> set_resource_path(resource_path)
+  end
+
+  @spec set_resource_path(Req.Request.t(), resource_path :: binary()) :: Req.Request.t()
+  def set_resource_path(req, resource_path) do
+    Req.merge(
+      req,
       resource_path: resource_path,
       resource_list_path: String.replace_suffix(resource_path, "/:name", "")
     )
   end
 
+  def discover_resource_path(_req, _group_version, _kind, _subresource) do
+    raise "Not implemented yet."
+  end
+
+  defmacro set_resource(req, group_version, kind, subresource \\ nil)
+           when is_binary(group_version) and is_binary(kind) do
+    discovery = elem(Code.eval_file("../build/resource_path_mapping.ex", __DIR__), 0)
+
+    case(discovery["#{group_version}/#{kind}/#{subresource}"]) do
+      nil ->
+        if subresource do
+          Logger.warning("""
+          The subresource "#{subresource}" of resource "#{kind}" in apiVersion "#{group_version}"
+          is not known at compile time. Kubereq is going to query the Cluster at runtime for this
+          resource. But if this is a "standard" Kubernetes resource, there might be a typo.
+          """)
+        else
+          Logger.warning("""
+          The resource "#{kind}" in apiVersion "#{group_version}" is not known at compile time.
+          Kubereq is going to query the Cluster at runtime for this resource. But if this is a
+          "standard" Kubernetes resource, there might be a typo.
+          """)
+        end
+
+        quote do
+          Kubereq.discover_resource_path(
```

Comment:
To be honest, I think having an extra request at runtime to discover a "known" path is not worth it.

Having a convenience for the built-in resources sounds good. I'd say it doesn't have to be a macro, but in this case the advantage of a macro is that we don't need to keep the mapping in memory at runtime.

Reply:
> advantage of a  macro is that we don't need to keep the mapping in memory at runtime

Yeah these were my thoughts exactly. Thanks for confirming that I'm not completely off course. :) 
But if I get you right, you don't value the compile-time warning too much?

Reply:
That's definitely subjective, but yeah, in this case I don't have strong preference for compile time warning/error. There are other subjects to typos like manifest keys and selectors, and the response body is also elaborate, so I expect people always need to play around with requests to get what they want, and so a reasonable runtime error works just as good.